Tailwind CSS language server doesn't escape space in path.

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

My username in Windows is Aldo Ridhoni. When opening a project with tailwindcss-language-enabled, a windows popup titled "Select an App to Open Aldo" is shown.

### Environment

Zed 0.147.0
Windows 11

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Icannot screenshot the Select an App to Open Aldo popup. 
Here is screenshot of powershell event log. 
![image](https://github.com/user-attachments/assets/1285dde4-ff78-4e71-a535-1fc127643e93)


### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
Caused by:
    The system cannot find the file specified. (os error 2)
2024-07-29T22:37:38.6097415+07:00 [INFO] starting language server "tailwindcss-language-server", path: "C:\\Users\\Aldo Ridhoni\\XXXX\\XXXX\\XXXX", id: 7
2024-07-29T22:37:38.610115+07:00 [INFO] starting language server. binary path: "powershell.exe", working directory: "C:\\Users\\Aldo Ridhoni\\XX\\XX\\XXX", args: ["C:\\Users\\Aldo Ridhoni\\AppData\\Local\\Zed\\languages\\tailwindcss-language-server\\node_modules/.bin/tailwindcss-language-server.ps1", "--stdio"]
2024-07-29T22:37:38.8569836+07:00 [ERROR] cannot read LSP message headers
2024-07-29T22:37:38.8572663+07:00 [ERROR] failed to start language server "tailwindcss-language-server": oneshot canceled
2024-07-29T22:37:38.8574055+07:00 [ERROR] server stderr: Some("")
2024-07-29T22:37:38.8575424+07:00 [INFO] retrying installation of language server "tailwindcss-language-server" in 1s
2024-07-29T22:37:38.8577201+07:00 [ERROR] server shut down
2024-07-29T22:37:39.861547+07:00 [ERROR] The system cannot find the path specified. (os error 3)
2024-07-29T22:37:39.8620616+07:00 [INFO] About to spawn test binary
2024-07-29T22:37:39.8624025+07:00 [WARN] test binary failed to launch
2024-07-29T22:37:39.8627131+07:00 [WARN] test binary check failed
2024-07-29T22:37:39.8630149+07:00 [INFO] beginning to reinstall server
2024-07-29T22:37:39.8633677+07:00 [INFO] deleting server container
2024-07-29T22:37:39.8640798+07:00 [ERROR] server container removal
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/languages/src/tailwind.rs
+++ b/crates/languages/src/tailwind.rs
@@ -119,14 +119,7 @@ impl LspAdapter for TailwindLspAdapter {
 
         #[cfg(target_os = "windows")]
         {
-            let mut env_path = vec![self.node.binary_path().await?.to_path_buf()];
-
-            if let Some(existing_path) = std::env::var_os("PATH") {
-                let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
-                env_path.append(&mut paths);
-            }
-
-            let env_path = std::env::join_paths(env_path)?;
+            let env_path = self.node.node_environment_path().await?;
             let mut env = HashMap::default();
             env.insert("PATH".to_string(), env_path.to_string_lossy().to_string());
 

--- a/crates/languages/src/tailwind.rs
+++ b/crates/languages/src/tailwind.rs
@@ -23,8 +23,26 @@ const SERVER_PATH: &str = "node_modules/.bin/tailwindcss-language-server.ps1";
 #[cfg(not(target_os = "windows"))]
 const SERVER_PATH: &str = "node_modules/.bin/tailwindcss-language-server";
 
+#[cfg(not(target_os = "windows"))]
 fn server_binary_arguments(server_path: &Path) -> Vec<OsString> {
     vec![server_path.into(), "--stdio".into()]
+}
+
+#[cfg(target_os = "windows")]
+fn server_binary_arguments(server_path: &Path) -> Vec<OsString> {
+    let x = vec![
+        "-File".into(),
+        // format!("{}", server_path.display())
+        //     .replace('/', "\\")
+        //     .into(),
+        server_path.into(),
+        "--stdio".into(),
+        // format!("& '{}' --stdio", server_path.display()).into(),
+    ];
+    // let x = "s".to_string()
+    println!("--> {:?}", x);
+    x
+    // vec![format!("\"& '{}' --stdio\"", server_path.display()).into()]
 }
 
 pub struct TailwindLspAdapter {
@@ -113,13 +131,7 @@ impl LspAdapter for TailwindLspAdapter {
 
         #[cfg(target_os = "windows")]
         {
-            let mut env_path = vec![self
-                .node
-                .binary_path()
-                .await?
-                .parent()
-                .expect("invalid node binary path")
-                .to_path_buf()];
+            let mut env_path = vec![self.node.binary_path().await?.to_path_buf()];
 
             if let Some(existing_path) = std::env::var_os("PATH") {
                 let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
@@ -130,6 +142,7 @@ impl LspAdapter for TailwindLspAdapter {
             let mut env = HashMap::default();
             env.insert("PATH".to_string(), env_path.to_string_lossy().to_string());
 
+            println!("--> {}", server_path.display());
             Ok(LanguageServerBinary {
                 path: "powershell.exe".into(),
                 env: Some(env),

--- a/crates/languages/src/tailwind.rs
+++ b/crates/languages/src/tailwind.rs
@@ -30,19 +30,7 @@ fn server_binary_arguments(server_path: &Path) -> Vec<OsString> {
 
 #[cfg(target_os = "windows")]
 fn server_binary_arguments(server_path: &Path) -> Vec<OsString> {
-    let x = vec![
-        "-File".into(),
-        // format!("{}", server_path.display())
-        //     .replace('/', "\\")
-        //     .into(),
-        server_path.into(),
-        "--stdio".into(),
-        // format!("& '{}' --stdio", server_path.display()).into(),
-    ];
-    // let x = "s".to_string()
-    println!("--> {:?}", x);
-    x
-    // vec![format!("\"& '{}' --stdio\"", server_path.display()).into()]
+    vec!["-File".into(), server_path.into(), "--stdio".into()]
 }
 
 pub struct TailwindLspAdapter {
@@ -142,7 +130,6 @@ impl LspAdapter for TailwindLspAdapter {
             let mut env = HashMap::default();
             env.insert("PATH".to_string(), env_path.to_string_lossy().to_string());
 
-            println!("--> {}", server_path.display());
             Ok(LanguageServerBinary {
                 path: "powershell.exe".into(),
                 env: Some(env),

--- a/crates/node_runtime/src/node_runtime.rs
+++ b/crates/node_runtime/src/node_runtime.rs
@@ -10,6 +10,7 @@ use semver::Version;
 use serde::Deserialize;
 use smol::io::BufReader;
 use smol::{fs, lock::Mutex, process::Command};
+use std::ffi::OsString;
 use std::io;
 use std::process::{Output, Stdio};
 use std::{
@@ -55,6 +56,7 @@ pub struct NpmInfoDistTags {
 #[async_trait::async_trait]
 pub trait NodeRuntime: Send + Sync {
     async fn binary_path(&self) -> Result<PathBuf>;
+    async fn node_environment_path(&self) -> Result<OsString>;
 
     async fn run_npm_subcommand(
         &self,
@@ -216,6 +218,22 @@ impl NodeRuntime for RealNodeRuntime {
         Ok(installation_path.join(NODE_PATH))
     }
 
+    async fn node_environment_path(&self) -> Result<OsString> {
+        let installation_path = self.install_if_needed().await?;
+        let node_binary = installation_path.join(NODE_PATH);
+        let mut env_path = vec![node_binary
+            .parent()
+            .expect("invalid node binary path")
+            .to_path_buf()];
+
+        if let Some(existing_path) = std::env::var_os("PATH") {
+            let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
+            env_path.append(&mut paths);
+        }
+
+        Ok(std::env::join_paths(env_path).context("failed to create PATH env variable")?)
+    }
+
     async fn run_npm_subcommand(
         &self,
         directory: Option<&Path>,
@@ -224,21 +242,9 @@ impl NodeRuntime for RealNodeRuntime {
     ) -> Result<Output> {
         let attempt = || async move {
             let installation_path = self.install_if_needed().await?;
-
             let node_binary = installation_path.join(NODE_PATH);
             let npm_file = installation_path.join(NPM_PATH);
-            let mut env_path = vec![node_binary
-                .parent()
-                .expect("invalid node binary path")
-                .to_path_buf()];
-
-            if let Some(existing_path) = std::env::var_os("PATH") {
-                let mut paths = std::env::split_paths(&existing_path).collect::<Vec<_>>();
-                env_path.append(&mut paths);
-            }
-
-            let env_path =
-                std::env::join_paths(env_path).context("failed to create PATH env variable")?;
+            let env_path = self.node_environment_path().await?;
 
             if smol::fs::metadata(&node_binary).await.is_err() {
                 return Err(anyhow!("missing node binary file"));
@@ -420,6 +426,10 @@ impl FakeNodeRuntime {
 #[async_trait::async_trait]
 impl NodeRuntime for FakeNodeRuntime {
     async fn binary_path(&self) -> anyhow::Result<PathBuf> {
+        unreachable!()
+    }
+
+    async fn node_environment_path(&self) -> anyhow::Result<OsString> {
         unreachable!()
     }
 

